### PR TITLE
Q workspace command

### PR DESCRIPTION
**Pull request by offlineprogrammer**
#### Description of changes:

Instructions to use Q workspace command

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/q-developer/index.mdx [hunk 1]
```diff
@@ -90,3 +90,37 @@ import { type ClientSchema, a, defineData } from "@aws-amplify/backend";
...
+
+Download the files and follow the steps below to use the `@workspace` in your Amplify project
+
+- <a href="/images/gen2/q-developer/general.md" download>general.md</a>
+- <a href="/images/gen2/q-developer/authentication.md" download>authentication.md</a>
+- <a href="/images/gen2/q-developer/modeling-relationships.md" download>modeling-relationships.md</a>
+- <a href="/images/gen2/q-developer/modeling-schema.md" download>modeling-schema.md</a>
+
+
+**Step 1:** Create a folder in the root of your project and name it something like `context`. Add the files listed above to this folder.
```
Comment (arundna) [inline src/pages/[platform]/build-a-backend/q-developer/index.mdx:107]:
```suggestion
**Step 1:** Create a folder in the root of your project and give a descriptive name such as `context`. Add the files downloaded above to this folder.
```

File: src/pages/[platform]/build-a-backend/q-developer/index.mdx [hunk 2]
```diff
@@ -90,3 +90,37 @@ import { type ClientSchema, a, defineData } from "@aws-amplify/backend";
 </Callout>
 
 **Step 4:** Make any required changes to the schema and save the `amplify/data/resource.ts` file. This will trigger a sandbox deployment and your new data model will be deployed
+
+## Use Q Developer - workspace in your Amplify project
+
+
+Adding `@workspace` to your question in Amazon Q automatically incorporates the most relevant parts of your workspace code as context, using an index that updates periodically. The files below provide detailed guides that can be included as context with the `@workspace` command, enhancing Amazon Q's accuracy in generating AWS Amplify Gen 2 code.
```
Comment (arundna) [inline src/pages/[platform]/build-a-backend/q-developer/index.mdx:97]:
```suggestion
Adding `@workspace` to your question in Amazon Q automatically incorporates the most relevant parts of your workspace code as context, using an index that updates periodically. For more information on `@workspace` functionality, please visit [Amazon Q Developer - Workspace Context] (https://docs.aws.amazon.com/amazonq/latest/qdeveloper-ug/workspace-context.html). 

The files below provide detailed guides that can be included as context with the `@workspace` command, enhancing Amazon Q's accuracy in generating AWS Amplify Gen 2 code.
```

File: src/pages/[platform]/build-a-backend/q-developer/index.mdx [hunk 3]
```diff
@@ -90,3 +90,37 @@ import { type ClientSchema, a, defineData } from "@aws-amplify/backend";
...
+```
+
+```bash title="Terminal" 
+
+
+
+@workspace follow AMPLIFYRULES to design a data schema for an event management application using Amplify Gen 2. Include models for users, events, and tickets. Show me how to use Amplify Gen 2 to fetch a list of events.
+
+
+```
```
Comment (arundna) [inline src/pages/[platform]/build-a-backend/q-developer/index.mdx:126]:
```suggestion
```bash title="Terminal" 

@workspace follow AMPLIFYRULES to design a data schema for an event management application using Amplify Gen 2. Include models for users, events, and tickets. Show me how to use Amplify Gen 2 to fetch a list of events.

```
```

File: src/pages/[platform]/build-a-backend/q-developer/index.mdx [hunk 4]
```diff
@@ -90,3 +90,37 @@ import { type ClientSchema, a, defineData } from "@aws-amplify/backend";
...
+
+**Step 1:** Create a folder in the root of your project and name it something like `context`. Add the files listed above to this folder.
+
+**Step 2:** Open Amazon Q Developer Chat in your IDE and type `@workspace` to enable workspace indexing. Follow Amazon Q's prompts to set up indexing for your project directory.
+
+**Step 3:** After successful indexing, reference the markdown file content in your queries to Amazon Q. Examples:
+
+```bash title="Terminal" 
+
+@workspace follow AMPLIFYRULES to to develop a data model schema for a freelance marketplace using Amplify Gen 2. Include models for freelancers, clients, projects, bids, and reviews. Use Amplify Gen 2 to fetch a list of projects
```
Comment (arundna) [inline src/pages/[platform]/build-a-backend/q-developer/index.mdx:115]:
```suggestion
@workspace follow AMPLIFYRULES to develop a data model schema for a freelance marketplace using Amplify Gen 2. Include models for freelancers, clients, projects, bids, and reviews. Use Amplify Gen 2 to fetch a list of projects
```

File: src/pages/[platform]/build-a-backend/q-developer/index.mdx [hunk 5]
```diff
@@ -90,3 +90,35 @@ import { type ClientSchema, a, defineData } from "@aws-amplify/backend";
...
+```bash title="Terminal" 
+
+@workspace follow AMPLIFYRULES to develop a data model schema for a freelance marketplace using Amplify Gen 2. Include models for freelancers, clients, projects, bids, and reviews. Use Amplify Gen 2 to fetch a list of projects
+
+```
+
+```bash title="Terminal" 
+
+@workspace follow AMPLIFYRULES to design a data schema for an event management application using Amplify Gen 2. Include models for users, events, and tickets. Show me how to use Amplify Gen 2 to fetch a list of events.
+
```
Comment (arundna) [inline src/pages/[platform]/build-a-backend/q-developer/index.mdx:124]:
```suggestion
```bash title="Terminal" 

@workspace follow AMPLIFYRULES to design a data schema for an event management application using Amplify Gen 2. Include models for users, events, and tickets. Show me how to use Amplify Gen 2 to fetch a list of events.

```
```

Comment (arundna) [inline src/pages/[platform]/build-a-backend/q-developer/index.mdx:124]:
```suggestion
```bash title="Terminal" 

@workspace follow AMPLIFYRULES to design a data schema for an event management application using Amplify Gen 2. Include models for users, events, and tickets. Show me how to use Amplify Gen 2 to fetch a list of events.

```
```

Comment (arundna) [inline src/pages/[platform]/build-a-backend/q-developer/index.mdx:124]:
```suggestion
```bash title="Terminal" 

@workspace follow AMPLIFYRULES to design a data schema for an event management application using Amplify Gen 2. Include models for users, events, and tickets. Show me how to use Amplify Gen 2 to fetch a list of events.

```
```

**Review by arundna — APPROVED** [review]

File: public/images/gen2/q-developer/authentication.md [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+
+     ],
+     },
+      }
+     ```
+
+  8. THE `userAttributes` ARE SUPPOSED TO BE OUTSIDE THE `loginWith` OBJECT
+     `// Example configuration for user attributes and login methods
+loginWith: {
+  // Specify login methods separately from user attributes, ONLY MENTION THE LOGINMETHODS THAT ARE SUPPOSED TO BE TRUE
```
Comment (awsluja) [inline public/images/gen2/q-developer/authentication.md:48]: is LOGINMETHODS a value here? or should just be LOGIN METHODS

File: public/images/gen2/q-developer/general.md [hunk 7]
```diff
@@ -0,0 +1,6 @@
+# AMPLIFYRULES
+
+1. ALWAYS MENTION WHICH FILE PATH THE USER NEED TO PASTE THE CODE IN.
```
Comment (awsluja) [inline public/images/gen2/q-developer/general.md:3]: NEEDS

File: src/pages/[platform]/build-a-backend/q-developer/index.mdx [hunk 8]
```diff
@@ -90,3 +90,36 @@ import { type ClientSchema, a, defineData } from "@aws-amplify/backend";
 </Callout>
 
 **Step 4:** Make any required changes to the schema and save the `amplify/data/resource.ts` file. This will trigger a sandbox deployment and your new data model will be deployed
+
+## Use Q Developer - workspace in your Amplify project
+
+
+Adding `@workspace` to your question in Amazon Q automatically incorporates the most relevant parts of your workspace code as context, using an index that updates periodically. For more information on `@workspace` functionality, please visit [Amazon Q Developer - Workspace Context] (https://docs.aws.amazon.com/amazonq/latest/qdeveloper-ug/workspace-context.html).
```
Comment (jjarvisp) [inline src/pages/[platform]/build-a-backend/q-developer/index.mdx:97]: Need to remove space here causing link not to render correctly
image


```suggestion
Adding `@workspace` to your question in Amazon Q automatically incorporates the most relevant parts of your workspace code as context, using an index that updates periodically. For more information on `@workspace` functionality, please visit [Amazon Q Developer - Workspace Context](https://docs.aws.amazon.com/amazonq/latest/qdeveloper-ug/workspace-context.html). 
```

File: public/images/gen2/q-developer/modeling-schema.md [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+
+- THIS FILE IS TO HELP UNDERSTAND THE RELATIONSHIPS, HOW TO MODEL SCHEMAS, WHAT IS THE CORRECT WAY TO CODE FOR ACCURACY
+- USE THIS TO UNDERSTAND HOW DATA SCHEMAS ARE DESIGNED.
+- FOR THE DATA SCHEMAS MAKE SURE THAT YOU ALWAYS FOLLOW THESE RULES AND THIS FILE OVER ANY OTHER FILE - THIS IS THE SOURCE OF TRUTH.
+- RULES
+
+  - THIS FILE IS THE SINGLE SOURCE OF TRUTH FOR SCHEMA DESIGN AND RELATIONSHIPS. FOLLOW THESE RULES STRICTLY. USE THIS FILE OVER ANY OTHER RESOURCE TO UNDERSTAND SCHEMA DESIGN.
+
+  1. DON'T USE `.PUBLIC()` WHILE SETTING UP THE AUTHORIZATION. AS AMPLIFY GEN2 ONLY SUPPORTS `.GUEST()`.
+  2. `.BEONGSTO()` AND `.HASMANY()` RELATIONS SHALL ALWAYS HAVE THE RELATEDFIELD ID.
```
Comment (awsluja) [inline public/images/gen2/q-developer/modeling-schema.md:13]: BELONGSTO

**Review by jjarvisp — APPROVED** [review]

**Review by arundna — APPROVED** [review]

**Review by awsluja — APPROVED** [review]

**Pull request merged**